updating/updating-disconnected-cluster.adoc

```diff
@@ -1,23 +1,150 @@
+
+. Mirror the version images to the internal container registry.
+
+** Connect the removable media to a system connected to the internet.
+** Mirror the images and configuration manifests to a directory on the removable media.
++
+----
+$ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/${RELEASE_NAME}:${OCP_RELEASE}-${ARCHITECTURE} --dry-run
+----
++
+----
+$ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/${RELEASE_NAME}:${OCP_RELEASE}-${ARCHITECTURE}
+----
++
+[NOTE]
+====
+If the local container registry and the cluster are connected to the mirror host, you can directly push the release images to the local registry and apply the ConfigMap to the   cluster using following command.
+----
+$ oc adm release mirror --from=quay.io/openshift-release-dev/${RELEASE_NAME}:${OCP_RELEASE}-${ARCHITECTURE} \
+  --to=${LOCAL_REGISTRY}/${LOCAL_REPOSITORY} --apply-release-image-signature
+----
+====
++
+** Take the media to the restricted network environment and upload the images to the local container registry.
++
+----
+$ oc image mirror  -a ${LOCAL_SECRET_JSON} "--from-dir=${REMOVABLE_MEDIA_PATH}/mirror" file://openshift/release:${OCP_RELEASE}* ${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}
+----
+
+=== Create the ConfigMap for image signature verification
+
+The Cluster Version Operator (CVO) in the cluster performs a verification to make sure the signatures of the release images are not tampered. To do that it compares expected VS actual image signatures. For the verification process it needs to know the image signatures. The image signatures are available in link:https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release[mirror.openshift.com]  and link:https://storage.googleapis.com/openshift-release/official/signatures[Google Cloud Storage (GCS)]. However in a restricted network environment we do not have access to the internet, so we need to create ConfigMap which CVO can use to verify the image signatures.
+
+[NOTE]
+====
+If you are upgrading from an earlier release than 4.4.8, you must must use the manual method for creating the ConfigMap.
+
+This is because the `oc` command-line interface (CLI) enhancement is not available in previous releases.
+====
+
+.Procedure
+
+. Use `oc` command-line interface (CLI) to log in to a given cluster which going to be upgraded.
+
++
+. Apply the the mirrored release image signature ConfigMap to the connected cluster
++
+----
+$ oc apply -f mirror/config/signature-sha256-81154f5c03294534.yaml
+----
+
+=== Perform the upgrade
+
+.Procedure
+
+. Perform the upgrade
++
+----
+$ oc adm upgrade --allow-explicit-upgrade --to-image ${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}@sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+----
+[NOTE]
+====
+You can use ImageContentSourcePolicy if you do want to use canonical registry name instead of LOCAL_REGISTRY.
```

Comment:
The idea is to communicate that if you are using `ImageContentSourcePolicy` you can use canonical registry names and I believe it communicates that. let me know if you disagree.

Reply:
Maybe my previous comment is not clear enough. When user mirror release image to a different PRODUCT_REPO for this upgrade with the source PRODUCT_REPO for disconnected install, then user have to capture `ImageContentSourcePolicy` from --dry-run output, and create it before upgrade:
https://github.com/openshift/openshift-docs/pull/21993/files#diff-8afabd3ee5993f3a7b9595c12f39a3baR88

For example:
1. mirror release image to 'release-4.3/ocp' in the container registry
2. run a disconnected install using the 'release-4.3/ocp' release image.
3. mirror release image to 'release-4.4/ocp' in the same container registry for disconnected cluster upgrade

In this case, user have to create `ImageContentSourcePolicy` before upgrade.

Reply:
> In this case, user MUST create `ImageContentSourcePolicy` before upgrade, or else, upgrade would fail.

You mean it will fail if user used quay.io as registry name right?

Reply:
The idea is to capture the good path in the doc and leave enough pointers to users to address their workflow. Also PRs are welcome if you want to improve the doc post this PR merge.

Reply:
@wking WDYT?

Reply:
> You mean it will fail if user used quay.io as registry name right?
As far as I know, this is not related to registry name, but PRODUCT_REPO path.

Reply:
Folks will need an ICSP so their cluster knows to check the mirror(s) to find the quay.io referenced images.  They may already have a sufficient ICSP in place, and we have no generic tooling to help users decide how/when to merge, dedup, and retire ICSPs.  I'm fine calling out tge "if you mirror all releases to the same place, you don't need to update your original ICSP" case.  But I don't think we want to go very far down the "I mirror each release to a different place" rabbit hole.

Reply:
Yeah it is tricky to explain it and at the same time not fall in to the trap of rabbit hole of explain how user needs to adjust the `ImageContentSourcePolicy`

Reply:
@jianlinliu I have made some changes, also tried the steps at https://gist.github.com/LalatenduMohanty/dcecc4dda52fcce6d17b4a6ef1249043 . Let me know if it looks fine to you.

Reply:
In QE's testing, we never use OpenShift Registry to host mirror release image. A registry running in a OCP cluster, which maybe also need a upgrade using a mirror release image hosted in itself, sound like not productive enough or not a customer case. 
Beside that, the steps in the gist looks good to me (Actually I did not find many difference with the existing steps in this doc PR).